src/componentarray.jl

```diff
@@ -138,12 +138,18 @@ make_carray_args(::NamedTuple{(), Tuple{}}) = (Any[], FlatAxis())
 make_carray_args(::Type{T}, ::NamedTuple{(), Tuple{}}) where {T} = (T[], FlatAxis())
 function make_carray_args(nt)
     data, ax = make_carray_args(Vector, nt)
-    data = length(data)==1 ? [data[1]] : reduce(vcat, data)
+    data = length(data)==1 ? [data[1]] : eltype(data) == Any ? reduce(vcat, data) : data
```

Comment:
This was kind of a lazy way I had to reset the eltype, but you're right, it's much better to do it a different way. Actually, I wonder if it would be better to use `recursive_type` here and do the promotion directly instead of doing the goofy `reduce` thing I was doing.

Reply:
I have remove the `reduce` here